### PR TITLE
chore: fix windows tests in 10.0

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### User facing changelog

n/a

### Additional details

Additional tests needed fixing in windows:

* The fuzzy spec search does not consistently use the windows separator. This PR ensures that we do.

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/frontend-shared/src/utils/spec-utils.ts
Comment (ryanthemanuel) [inline packages/frontend-shared/src/utils/spec-utils.ts:11]: This seems to be necessary since the specs list exists before Cypress necessarily exists

File: packages/frontend-shared/src/utils/spec-utils.ts
Comment (lmiller1990) [inline packages/frontend-shared/src/utils/spec-utils.ts:9]: I don't think we should use `__CYPRESS_CONFIG__` in `frontend-shared`, it's not available in launchpad - it's `app` specific.

How about exporting this function and using it? https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/src/runner/index.ts#L341-L343

File: packages/frontend-shared/src/utils/spec-utils.ts
Comment (lmiller1990) [inline packages/frontend-shared/src/utils/spec-utils.ts:11]: Actually this entire file should not be in `frontend-shared`, it's only used in `app`. How about moving it?

**Review by lmiller1990 — CHANGES_REQUESTED** [review]
Good fix, we move this entire file (and associated change) to `app`, and use the existing function to decode the __CYPRESS_CONFIG_ (see comment).

File: packages/app/src/specs/spec-utils.ts
Comment (lmiller1990) [inline packages/app/src/specs/spec-utils.ts:9]:
```suggestion
const platform = getRunnerConfigFromWindow().platform
```

I think - `window.Cypress` won't exist here, only on the runner page, but `__CYPRESS_CONFIG__` is defined from the very moment the app is opened (see here  https://github.com/cypress-io/cypress/blob/10.0-release/packages/data-context/src/sources/HtmlDataSource.ts#L119).

**Review by flotwig — APPROVED** [review]

**Review by ZachJW34 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Looks good... once CI is green, let's merge. I'll leave it to you 👍

**Review by lmiller1990 — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**
